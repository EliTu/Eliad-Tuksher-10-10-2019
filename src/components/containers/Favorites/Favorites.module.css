@import '../../../variables.css';

.FavoritesStyles {
    width: 90%;
    height: 600px;
    margin: 1.5rem auto;
}

.noFavoritesMessage {
    width: 80%;
    font-size: var(--text-size-large);
    margin: 1.5rem auto;
}


/* Medium screen */
@media only screen and (max-width: 800px) {
    .FavoritesStyles {
        width: 95%;
    }
}