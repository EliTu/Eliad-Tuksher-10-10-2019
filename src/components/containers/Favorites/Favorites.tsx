import React from 'react';
import { connect } from 'react-redux';
import FavoritesList from './FavoritesList/FavoritesList';
import styles from './Favorites.module.css';

interface IProps {
	isLoading: boolean;
	error: string;
}

const Favorites: React.FC<IProps> = ({ isLoading, error }) => {
	const { FavoritesStyles } = styles;
	return (
		<div className={FavoritesStyles}>
			{!isLoading && !error && <FavoritesList />}
		</div>
	);
};
// Redux setup:
const mapStateToProps = (state: any) => {
	return {
		isLoading: state.favorites.isLoading,
		error: state.currentWeather.error,
	};
};

export default connect(mapStateToProps)(Favorites);
