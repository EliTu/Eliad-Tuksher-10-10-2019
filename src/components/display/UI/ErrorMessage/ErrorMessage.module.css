@import '../../../../variables.css';

.ErrorMessageStyles {
    display: block;
    width: 50%;
    padding: 1rem 2rem;
    color: var(--color-main-black);
}

.ErrorMessageStyles p:nth-child(1) {
    font-size: var(--text-size-extra-large);
    font-weight: 800;
}

.ErrorMessageStyles p:nth-child(2) {
    font-size: var(--text-size-large);
    font-size: 600;
}

.ErrorMessageStyles p:nth-child(3) {
    font-size: var(--text-size-medium);
}

.ErrorMessageStyles p:nth-child(4) {
    font-size: var(--text-size-medium);
}

.ErrorMessageStyles span {
    color: rgba(248, 129, 129, 0.822);
    font-weight: 700;
}