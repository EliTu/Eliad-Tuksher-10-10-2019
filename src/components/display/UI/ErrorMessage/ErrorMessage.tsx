import React from 'react';
import styles from './ErrorMessage.module.css';

interface IProps {
	errorDetails: string | undefined;
}

export const ErrorMessage: React.FC<IProps> = ({ errorDetails }) => {
	const { ErrorMessageStyles } = styles;

	return (
		<div className={ErrorMessageStyles}>
			<p>Oops...</p>
			<p>Something went wrong with the server request :(</p>
			<p>
				The reason seems to be: <span>{errorDetails}</span>
			</p>
			<p>Please try again in a short while!</p>
		</div>
	);
};

export default ErrorMessage;
