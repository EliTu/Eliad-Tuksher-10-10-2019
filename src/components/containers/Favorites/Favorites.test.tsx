import React from 'react';
import { shallow, ShallowWrapper } from 'enzyme';
import { Favorites } from './Favorites';
import ErrorMessage from '../../display/UI/ErrorMessage/ErrorMessage';
import initialShallowRender from '../../../utilities/test-utilities/initialShallowRender';

describe('Favorites component', () => {
	let component: ShallowWrapper;
	beforeEach(
		() => (component = shallow(<Favorites isLoading={false} error={''} />))
	);

	it('should render without errors', () => {
		initialShallowRender(component, 'FavoritesStyles');
	});

	it('should not display the component if isLoading is true', () => {
		component.setProps({ isLoading: true });

		expect(component).toMatchSnapshot();
		expect(component.find('div').children()).not.toExist();
	});

	it('should dispaly the ErrorMessage component if error is truthy', () => {
		component.setProps({ error: 'abc' });
		expect(component).toMatchSnapshot();
		expect(component.children()).toContainReact(
			<ErrorMessage errorDetails={'abc'} />
		);
	});
});
