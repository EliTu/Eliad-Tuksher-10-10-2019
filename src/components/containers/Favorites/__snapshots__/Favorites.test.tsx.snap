// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Favorites component should dispaly the ErrorMessage component if error is truthy 1`] = `
<div
  className="FavoritesStyles"
>
  <ErrorMessage
    errorDetails="abc"
  />
</div>
`;

exports[`Favorites component should not display the component if isLoading is true 1`] = `
<div
  className="FavoritesStyles"
/>
`;

exports[`Favorites component should render without errors 1`] = `
<div
  className="FavoritesStyles"
>
  <withRouter(Connect(FavoritesList)) />
</div>
`;
